ompi/mca/coll/tuned/coll_tuned_dynamic_file.c

```diff
@@ -176,11 +177,38 @@ int ompi_coll_tuned_read_rules_config_file (char *fname, ompi_coll_alg_rule_t**
 
                 msg_p = &(com_p->msg_rules[nms]);
 
-                if( (getnext (fptr, &MS) < 0) || (MS < 0) ) {
+                char *msg_topo = NULL;
+                if( getnext_string(fptr, &msg_topo) < 0 ) {
+                    OPAL_OUTPUT((ompi_coll_tuned_stream,"Could not read message size/name of a topo level for collective ID %ld com rule %d msg rule %d at around line %d\n", 
+                                 CI, ncs, nms, fileline));
+                    goto on_file_error;
+                }
+                
+                char *temp_str = strdup(msg_topo);
+                char *msg_size = strtok(temp_str, "@");
+                if (NULL == msg_size) {
                     OPAL_OUTPUT((ompi_coll_tuned_stream,"Could not read message size for collective ID %ld com rule %d msg rule %d at around line %d\n", CI, ncs, nms, fileline));
                     goto on_file_error;
                 }
+                MS = atol(msg_size);
```

Comment:
This logic needs more input validation - what does `atol("123abc")` return?

Also, previously the code was checking for negative values which is missing here.